tests/test_discovered_resources.py

```diff
@@ -92,33 +95,102 @@ async def test_discovered_resource_create_batch(server, client, agent, environme
+    # Resource repartition and expected filtering results:
+
+    #                                        |              FILTER
+    # discovered    managed   orphaned       |  TRUE        FALSE          NONE
+    # ---------------------------------------+-------------------------------------
+    #     R1            x                    |   x                           x
+    #     R2            x                    |   x                           x
+    #     R3                      x          |   x                           x
+    #     R4                      x          |   x                           x
+    #     R5                                 |                x              x
+    #     R6                                 |                x              x
+
+    discovered_resources = []
+    for i in range(1, 7):
+        rid = f"test::Resource[agent1,key{i}=key{i}]"
+        discovered_resources.append(
+            {
+                "discovered_resource_id": rid,
+                "values": {"value1": f"test{i}", "value2": f"test{i+1}"},
+                "managed_resource_uri": (
+                    f"/api/v2/resource/{rid}" if i <= 4 else None
+                ),  # Last 2 resources are not known to the orchestrator
+            }
+        )
+
+    result = await agent._client.discovered_resource_create_batch(environment, discovered_resources)
     assert result.code == 200
 
+    version1 = await clienthelper.get_version()
+    orphaned_resources = [
+        {
+            "id": ResourceVersionIdStr(f"{res['discovered_resource_id']},v={version1}"),
+            "values": res["values"],
+            "requires": [],
+            "purged": False,
+            "send_event": False,
+        }
+        for res in discovered_resources[2:-2]
+    ]
+    await clienthelper.put_version_simple(resources=orphaned_resources, version=version1)
+
+    # Create some Resources that are already managed:
+    version2 = await clienthelper.get_version()
+    managed_resources = [
+        {
+            "id": ResourceVersionIdStr(f"{res['discovered_resource_id']},v={version2}"),
+            "values": res["values"],
+            "requires": [],
+            "purged": False,
+            "send_event": False,
+        }
+        for res in discovered_resources[:2]
+    ]
+    await clienthelper.put_version_simple(resources=managed_resources, version=version2)
+
+    if apply_filter is None:
+        filter = None
+        expected_result = discovered_resources
+    else:
+        filter = {"managed": apply_filter}
+        if apply_filter:
+            expected_result = discovered_resources[:-2]
+        else:
+            expected_result = discovered_resources[-2:]
+
     result = await client.discovered_resources_get_batch(
         environment,
+        filter=filter,
     )
     assert result.code == 200
-    assert len(result.result["data"]) == 6
+    assert result.result["data"] == expected_result
```

Comment:
This test case will be faster if you don't parameterize it and just put the three different call to the `discovered_resources_get_batch()` endpoint under each other. If you did it to prevent code duplication, then you can just put a for-loop around this code that loops through the different values of `apply_filter`.